src/videojs-hls.js

```diff
@@ -1030,6 +1030,11 @@ videojs.Hls.translateMediaIndex = function(mediaIndex, original, update) {
     // let the media index be zero when there are no segments defined
     return 0;
   }
+  
+  if (original.mediaSequence === update.mediaSequence) {
+	// playlist sequence didn't change
+	return mediaIndex;
```

Comment:
Isn't this equivalent to the [computation of translatedMediaIndex](https://github.com/ntadej/videojs-contrib-hls/blob/playlist-fix/src/videojs-hls.js#L1041)? If `original.mediaSequence === update.mediaSequence`, then:

```
(mediaIndex + (original.mediaSequence - update.mediaSequence))
=> (mediaIndex + (0))
=> mediaIndex
```
The only case where that value is unused is if the resulting index is outside the scope of the current playlist. Were you seeing situations where `translatedMediaIndex === update.segments.length`?